x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticatorChain.java

```diff
@@ -240,7 +241,10 @@ private Authentication lookForExistingAuthentication(Authenticator.Context conte
             logger.error(() -> format("caught exception while trying to read authentication from request [%s]", context.getRequest()), e);
             throw context.getRequest().tamperedRequest();
         }
-        if (authentication != null && context.getRequest() instanceof AuthenticationService.AuditableRestRequest) {
+        // this throws if a new http request looks to be already authenticated
+        if (authentication != null
+            && (context.getRequest() instanceof AuthenticationService.AuditableHttpRequest)
+            && (((AuthenticationService.AuditableHttpRequest) context.getRequest()).request instanceof HttpRequest == false)) {
```

Comment:
Right now, `HttpRequest`s are checked for authentication in two places: the original `SecurityRestFilter` and the new  `HttpHeadersAuthenticator`, so this check here must account for this fact. But it's likely that the final PR drops the `SecurityRestFilter`.